### PR TITLE
support for mesh powerstrap

### PR DESCRIPTION
**Pull request by yolowiz**
**Related PRs / Issues**

**Type of change**:
- [ ] Bug fix
- [x] New feature
- [ ] Other enhancement

**Impact**:
- [x] Change to core Hammer
- [ ] Change to a Hammer plugin
- [ ] Other

**Contributor Checklist**:
- [x] Did you set `master` as the base branch?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you update the `poetry.lock` file if you updated the requirements in `pyproject.toml`?
- [x] (If applicable) Did you add a unit test demonstrating the PR?
- [ ] (If applicable) Did you run this through the e2e integration tests?
- [ ] (If applicable) Did you update the submodules in `e2e/` if this feature depends on updated plugins?

### CODE REVIEW

**Review by harrisonliew — CHANGES_REQUESTED** [review]
Can you update the documentation with a diagram using this new method? I'd like to see whether the utilization property makes sense (recommend 0.2 for diagram).

File: doc/Hammer-Use/mesh.png
Comment (harrisonliew) [inline doc/Hammer-Use/mesh.png:1]: I'm confused about this picture. Isn't only the bottom half actually the "mesh" method? The top half is existing "coupled" with track_spacing = 1.

Comment (yolowiz) [reply doc/Hammer-Use/mesh.png:1]: It's an illustration of how twt is used.

Comment (harrisonliew) [reply doc/Hammer-Use/mesh.png:1]: In my opinion, that is unnecessary (you can just overlay measurement e in the bottom half).

If anything, the existing diagrams above this should indicate the differences in calculation between twwt and twt when track_spacing = 0 or > 0.

File: doc/Hammer-Use/mesh_description.png
Comment (harrisonliew) [inline doc/Hammer-Use/mesh_description.png:1]: Much better. But isn't there a slight error for mesh straps? If it's 50% utilization, we should see:
 VDD | signal | VSS | signal | VDD | signal ...
Right now there are 2 signals between each power => 33% utilization

Comment (yolowiz) [reply doc/Hammer-Use/mesh_description.png:1]: That arises from the fact that power utilization is not exact for power strap calculation: track_spacing is part of the consumed_tracks, and for mesh, track_spacing is hard-coded to 1.

Comment (harrisonliew) [reply doc/Hammer-Use/mesh_description.png:1]: Oh, this is your hack to force twt calculation, right? can you force that calculation without needing to account for the extra spacing?

Comment (yolowiz) [reply doc/Hammer-Use/mesh_description.png:1]: To keep the power utilization consistent with twt, yes. I can try to subtract the track_spacing after twt returns the width and spacing.

Comment (harrisonliew) [reply doc/Hammer-Use/mesh_description.png:1]: Yes, please subtract that extra track_spacing in mesh mode so that the calculation is intuitive.

**Review by harrisonliew — APPROVED** [review]

**Pull request merged**